ui/perfherder/compare/CompareTable.jsx

```diff
@@ -28,24 +45,173 @@ export default class CompareTable extends React.PureComponent {
+      // so they need to be initialized with a default value
+      if (shouldBeNumber) {
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(a[key]) && !Array.isArray(a[key])) a[key] = 0;
+        // eslint-disable-next-line no-restricted-globals
+        if (isNaN(b[key]) && !Array.isArray(b[key])) b[key] = 0;
+      } else {
+        a[key] = typeof a[key] === 'undefined' ? '' : a[key];
+        b[key] = typeof b[key] === 'undefined' ? '' : b[key];
+      }
+      const ak = Array.isArray(a[key]) ? a[key].length : a[key];
+      const bk = Array.isArray(b[key]) ? b[key].length : b[key];
+      if (ak < bk) {
+        if (sortOrderAsc) return -1;
+        return 1;
+      }
+      if (ak > bk) {
+        if (sortOrderAsc) return 1;
+        return -1;
+      }
+      return 0;
+    };
+  };
+
+  sortBy = (key, data) => {
+    const { sortOrderAsc } = this.state
+    let {
+      borderTopColor,
+      borderTopWidth,
+      borderBottomColor,
+      borderBottomWidth,
+    } = this.state;
+    const columnName = key;
+    // every other columns except "name" are numbers
+    const shouldBeNumber = key !== 'name';
+    data.sort(this.compareBy(key, shouldBeNumber));
+    const sorted = true;
+    if (sorted) {
+      if (sortOrderAsc) {
+        borderTopColor = '#333';
+        borderTopWidth = '1px';
+        borderBottomColor = '';
+        borderBottomWidth = '';
+        this.setState({
+          columnName,
+          sortOrderAsc: !sortOrderAsc,
+          sorted,
+          borderTopColor,
+          borderTopWidth,
+          borderBottomColor,
+          borderBottomWidth,
+        });
+      } else {
+        borderBottomColor = '#333';
+        borderBottomWidth = '1px';
+        borderTopColor = '';
+        borderTopWidth = '';
+        this.setState({
+          columnName,
+          sortOrderAsc: !sortOrderAsc,
+          sorted,
+          borderTopColor,
+          borderTopWidth,
+          borderBottomColor,
+          borderBottomWidth,
+        });
+      }
+    };
+    // console.log(`sortOrderAsc: ${sortOrderAsc}`);
+    // console.log(`sortBy sorted: ${sorted}`);
+
+    this.forceUpdate();
+  };
+
   render() {
+    // eslint-disable-next-line react/prop-types
     const { data, testName } = this.props;
+    const {
+      columnName,
+      sorted,
+      borderTopColor,
+      borderTopWidth,
+      borderBottomColor,
+      borderBottomWidth,
+    } = this.state;
+    console.log(`data: ${JSON.stringify(data)}`);
+    console.log(`render sorted: ${JSON.stringify(sorted)}`);
     return (
       <Table sz="small" className="compare-table mb-0 px-0" key={testName}>
         <thead>
-          <tr className="subtest-header bg-lightgray">
-            <th className="text-left">
-              <span>{testName}</span>
+          <tr className={`${sorted ? 'subtest-header-visible ' : 'subtest-header '}bg-lightgray`}>
+            <th
+              style={{
```

Comment:
This style thing repeats the same 4 properties for all the sortable headings. Not sure how I can reduce this redundancy without major changes.